src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -328,9 +328,11 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_mpi_put(const void *origin_addr,
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_NM_MPI_PUT);
     int mpi_errno = MPI_SUCCESS;
 
-    if (!MPIDI_OFI_ENABLE_RMA || !(winattr & MPIDI_WINATTR_NM_REACHABLE)) {
-        mpi_errno = MPIDIG_mpi_put(origin_addr, origin_count, origin_datatype, target_rank,
-                                   target_disp, target_count, target_datatype, win);
+    if (!MPIDI_OFI_ENABLE_RMA || !(winattr & MPIDI_WINATTR_NM_REACHABLE) ||
+        MPL_gpu_query_pointer_is_dev(origin_addr)) {
+        mpi_errno =
```

Comment:
Do we need check also for FOP and CSWAP?

Reply:
Yes, probably. I only touched put/get because I started with ucx. I will update tomorrow to address ch4/ofi atomics.